### ISSUE TITLE
Two maps are now working inside dbc.Tab component

### ISSUE THREAD

**Issue by SergienkoSV** [thedirtyfew/dash-leaflet#58]
Hi,
I'm building an interactive dashboard which using some maps. 
I'd like to put two separate maps to two tabs inside `dbc.Tab` container. However, just one tile layer works as expected, second has incorrect tiles and doen't responce properly to map movement.

```python
# Here's a minimal example that reproduces the error
import dash
import dash_bootstrap_components as dbc
import dash_leaflet as dl

app = dash.Dash(external_stylesheets=[dbc.themes.BOOTSTRAP])
tab1 = dbc.Tab(dl.Map([dl.TileLayer()], id='map1', style={'width': '1000px', 'height': '500px'}), label="Tab1", tab_id="tab1")
tab2 = dbc.Tab(dl.Map(dl.TileLayer(), id='map2', style={'width': '800px', 'height': '300px'}), label="Tab2", tab_id="tab2")
app.layout = dbc.Container(dbc.Tabs([tab1, tab2], active_tab='tab1', id='tabs'))

if __name__ == '__main__':
    app.run_server(debug=True)

```

Any ideas?

**Comment by SergienkoSV** [comment]
Tab2 tiles looks like this (Tab1 tiles are ok):

![image](https://user-images.githubusercontent.com/23247173/99920213-0d9e6b00-2d33-11eb-8280-b0df8d56dcd7.png)

**Comment by SergienkoSV** [comment]
I'm using dash-leaflet-0.1.8
Is this similar with:
https://github.com/thedirtyfew/dash-leaflet/issues/56

**Comment by emilhe** [comment]
I think it's because the map is rendered before it's shown (similar to [this issue](https://github.com/thedirtyfew/dash-leaflet/issues/36)). One way to solve the problem would be to render the content dynamically, i.e. something like

```
import dash
import dash_bootstrap_components as dbc
import dash_html_components as html
import dash_leaflet as dl
from dash.dependencies import Input, Output

app = dash.Dash(external_stylesheets=[dbc.themes.BOOTSTRAP])
app.layout = html.Div(
    [
        dbc.Tabs(
            [
                dbc.Tab(label="Tab 1", tab_id="tab-1"),
                dbc.Tab(label="Tab 2", tab_id="tab-2"),
            ],
            id="tabs",
            active_tab="tab-1",
        ),
        html.Div(id="content"),
    ]
)

@app.callback(Output("content", "children"), [Input("tabs", "active_tab")])
def switch_tab(at):
    if at == "tab-1":
        return dl.Map(id="map1",children=[dl.TileLayer(id="tile1")], style={'width': '1000px', 'height': '500px'})
    elif at == "tab-2":
        return dl.Map(id="map2",children=[dl.TileLayer(id="tile2")],center=[40, -100], zoom=4, style={'width': '1000px', 'height': '500px'})
    return html.P("This shouldn't ever be displayed...")

if __name__ == '__main__':
    app.run_server(debug=True)
```

**Comment by emilhe** [comment]
I believe this issue is fixed in 0.1.12, please reopen if that is not the case.

**Comment by PeterQFR** [comment]
for issues where the rendered tab also has other components also containing callbacks, I wasn't able to get the above to work. Mainly because this doesn't create the content until the tab is selected which prevents callbacks from being initialised. 

In that case I found that rendering my views in their own divisions and hiding the one I didn't want based on the active tab works and preserves the callback initialisaiton at the start of the app:
```
import dash
import dash_bootstrap_components as dbc
import dash_html_components as html
import dash_leaflet as dl
from dash.dependencies import Input, Output

app = dash.Dash(external_stylesheets=[dbc.themes.BOOTSTRAP])

taba = html.Div(createCardAWithCallbacks(), hidden=False, id='taba')
tabb=html.Div(createCardBWithCallbacks(), hidden=False, id='tabb')

app.layout = html.Div(
    [
        dbc.Tabs(
            [
                dbc.Tab(label="Tab 1", tab_id="tab-1"),
                dbc.Tab(label="Tab 2", tab_id="tab-2"),
            ],
            id="tabs",
            active_tab="tab-1",
        ),
        html.Div(id="content"),
        taba,
        tabb
    ]
)

@app.callback(Output("taba", "hidden"), [Input("tabs", "active_tab")])
def switch_tab(at):
    if at == "tab-0":
        return False
    elif at == "tab-1":
        return True
    return html.P("This shouldn't ever be displayed...")

@app.callback(Output("tabb", "hidden"), [Input("tabs", "active_tab")])
def switch_tab(at):
    if at == "tab-0":
        return True
    elif at == "tab-1":
        return False
    return html.P("This shouldn't ever be displayed...")

if __name__ == '__main__':
    app.run_server(debug=True)
```

In this the two functions `createCardAWithCallbacks()` and `createCardBWithCallbacks()` return a list of children representing buttons, the dl.Map and other UI elements which may have callbacks associated with them (not shown)

It works as it allows the callbacks to be initialised, also renders the maps correctly and the hiding of the html.Div, doesn't seem to mess with that.
